app/ante/eth.go

```diff
@@ -218,6 +217,9 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 
 	// Set ctx.GasMeter with a limit of GasWanted (gasLimit)
 	gasConsumed := ctx.GasMeter().GasConsumed()
+	// We can't trust the tx gas limit, because we'll refund the unused gas.
+	// Validator should set a relatively small mempool size to further protect against DoS attack.
+	gasWanted := uint64(len(tx.GetMsgs())) * DefaultTxGasWanted
```

Comment:
I think we should set `gasWanted` as the min between the tx gas limit and the default

Reply:
I think `21000` is the minimal gas requirement of eth tx.

Reply:
I renamed it to `MinimalTxGasWanted` to make it clearer.